onnxruntime/core/session/ep_plugin_provider_interfaces.cc

```diff
@@ -0,0 +1,270 @@
+    return {};
+  }
+
+  OrtEpGraphSupportInfo api_graph_support_info(*ep_graph);
+  UniqueOrtStatus status(ort_ep_->GetCapability(ort_ep_.get(), ep_graph->ToExternal(), &api_graph_support_info),
+                         OrtApis::ReleaseStatus);
+
+  if (status != nullptr) {
+    LOGS_DEFAULT(ERROR) << "OrtEp::GetCapability() failed with error: " << ToStatus(status.get()).ToString();
+    return {};
+  }
+
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+  result.reserve(api_graph_support_info.node_groupings.size());
+  if (api_graph_support_info.node_groupings.empty()) {
+    return {};
+  }
+
+  ModelMetadefIdGenerator generator;
+
+  // Create ComputeCapability instances from OrtEpGraphSupportInfo::NodeGrouping instances.
+  for (const OrtEpGraphSupportInfo::NodeGrouping& node_grouping : api_graph_support_info.node_groupings) {
+    if (node_grouping.kind == OrtEpGraphSupportInfo::NodeGroupingKind::kSingleAssignedNode) {
+      auto indexed_sub_graph = std::make_unique<IndexedSubGraph>();
+
+      indexed_sub_graph->nodes.push_back(node_grouping.nodes[0]->node.Index());
+      result.push_back(std::make_unique<ComputeCapability>(std::move(indexed_sub_graph)));
+    } else if (node_grouping.kind == OrtEpGraphSupportInfo::NodeGroupingKind::kFusedNode) {
+      std::unordered_set<const Node*> node_set;
+      node_set.reserve(node_grouping.nodes.size());
+      for (const EpNode* ep_node : node_grouping.nodes) {
+        node_set.insert(&ep_node->node);
+      }
+
+      // We now require the OrtEp to only provide individual groups of supported nodes that each maps to exactly
+      // one ComputeCapability. Calling utils::CreateSupportedPartitions() may create multiple ComputeCapability
+      // instances, and if so, log an error and return.
+      std::vector<std::unique_ptr<ComputeCapability>> capabilities = utils::CreateSupportedPartitions(
+          graph_viewer, node_set, /*stop_ops*/ {}, PluginEpMetaDefNameFunctor(generator, graph_viewer, this->Type()),
+          this->Type(), this->Type(), /*node_unit_map*/ nullptr);
+
+      if (capabilities.size() > 1) {
+        LOGS_DEFAULT(ERROR) << "OrtEp::GetCapability() set nodes that cannot be fused together. "
+                            << "Please ensure that the nodes provided to EpGraphSupportInfo_AddFusedNodes() do not "
+                            << "have an unsupported node in any path between two of the supported nodes.";
+        return {};
+      }
+
+      for (auto& capability : capabilities) {
+        result.push_back(std::move(capability));
+      }
+    } else {
+      LOGS_DEFAULT(ERROR) << "PluginExecutionProvider::GetCapability() has invalid NodeGroupingKind: "
+                          << static_cast<int>(node_grouping.kind);
+      return {};
+    }
+  }
+
+  return result;
+}
+
+PluginExecutionProvider::FusedNodeState& PluginExecutionProvider::PushFusedNodeState(size_t num_fused_nodes) {
```

Comment:
nit: I expect this class will get quite big, so helpers like this could be local functions to minimize the class API.

Reply:
Moved this to a local function. Thanks.